tsc/frr/sonic_frr_upgrade_process.md

```diff
@@ -0,0 +1,77 @@
+# **SONiC Community FRR Upgrade Process Proposal**
+
+FRR upgrade and patch cadence is largely on need basis in the current SONiC release program. This SONiC FRR upgrade process proposal is to formalize the FRR upgrade cadence and process for future SONiC release programs.  
+
+# SONiC FRR Version Upgrade History
+
+<p align=center>
+<img src="img/frr.png" alt="">
+</p>
+
+
+# FRR Project Release Cadence
+-   FRR release numbering scheme x.y.z-s#
+-   New FRR releases roughly every 4 months
+	- Apr 2023 - Release 8.5.1 (48 fixes)
+	- Mar 2023 - Release 8.5 (947+ commits)
+	- Jan 2023 - Release 8.4.2 (22 fixes)
+	- Nov 2022 - Release 8.4.1 (16 fixes)
+	- Nov 2022 - Release 8.4 (700+ commits)
+	- Aug 2022 - Release 8.3.1 (14 fixes)
+	- Jul 2022 - Release 8.3 (1000+ commits)
+	- Mar 2022 - Release 8.2.2 (800+ commits)
+	- Nov 2021 - Release 8.1.0 (1200+ commits)
+	- Jul 2021 - Release 8.0.0 (2200+ commits)
+
+-   SONiC to stay out from major/minor releases (x.y) and use patch release (.z) for stability (eg, FRR 8.3.1 instead of 8.3 if it is for 202211 release)
+
+# SONiC Release FRR Upgrade
+-   SONiC default to rebase FRR in every November community release
+-   SONiC FRR upgrade test requirements
+	-   Pass FRR upgrade acceptance test (eg, 4 node BGP topo traffic test, Spytest sonic-vs L3 testsuite)
+	-   Pass existing Azure pipeline build test and LGTM
+-   Rotate SONiC FRR maintenance duty among repo maintainer org and others (BRCM, MSFT, Alibaba, NVDA, DELL)
+-   Responsibility of SONiC FRR release maintainer
+	-   Default 12 months assignment (or until the next upgrade)
+	-   Upgrade FRR version in Nov release, resolve SONiC FRR upgrade integration issues
+	-   Triage and fix SONiC FRR issues
+	-   Submit fixes to FRR project, submit new FRR topo test to FRR project if there is a gap
```

Comment:
Shouldn't it be the responsibility of FRR community to fix its issues rather than SONiC community working on fixes there? I believe the flow should be SONiC community members should raise the FRR issues in FRR GitHub found during SONiC testing but the responsibility to fix it should be in FRR community. Not all companies have resources with FRR expertise and hence this may be difficult for them to accomplish if they own FRR maintenance

Reply:
FRR issues found in SONiC integration maybe specific to SONiC use case, I can change the language not to mandate the maintainer to fix, eg, the fix can come from SONiC contributors or from FRR community, but the maintainer is responsible to drive the fix to unblock SONiC community